src/main/java/de/tum/in/www1/artemis/service/FeedbackService.java

```diff
@@ -148,6 +147,21 @@ public Feedback createFeedbackFromTestCase(String testName, List<String> testErr
      * @return A filtered and better formatted error message
      */
     private static String processResultErrorMessage(final ProgrammingLanguage programmingLanguage, final String message) {
+        final String timeoutDetailText = "The test case execution timed out. This indicates issues in your code such as endless for / while loops or issues with recursion. Please carefully review your code to avoid such issues. In case you are absolutely sure that there are no issues like this, please contact your instructor to check the setup of the test.";
+        // Overwrite timeout exception messages for Junit4, Junit5 and other
+        List<String> exceptions = Arrays.asList("org.junit.runners.model.TestTimedOutException", "java.util.concurrent.TimeoutException",
+                "org.awaitility.core.ConditionTimeoutException", "Timed?OutException");
+        Pattern findPattern = Pattern.compile("^.*(" + String.join("|", exceptions) + ").*");
+        Matcher matcher = findPattern.matcher(message);
+        if (matcher.find()) {
+            return matcher.group(1) + ": " + timeoutDetailText;
+        }
+        else if (message.contains("timed out after")) {
+            // overwrite AJTS: TimeoutException
+            return timeoutDetailText;
```

Comment:
We should keep the original message as it might contain additional details. I suggest that we use something like timeoutDetailText + ":\n " + message

Reply:
Good idea. 
Done in 38d920a53022f1d2692ed329f1a4159a422887ba